src/Examine.Core/OrderedDictionary.cs

```diff
@@ -89,7 +97,7 @@ public bool TryGetValue(TKey key, out TVal value)
                 {
                     return found.Value;
                 }
-                return default(TVal);
+                return default(TVal); // TODO: should this throw instead
```

Comment:
Yes, probably should throw

Reply:
This now throws a `KeyNotFoundException`